lighthouse-core/lib/page-functions.js

```diff
@@ -113,18 +113,24 @@ function getElementsInDocument(selector) {
  */
 /* istanbul ignore next */
 function getOuterHTMLSnippet(element, ignoreAttrs = []) {
-  const clone = element.cloneNode();
-
-  ignoreAttrs.forEach(attribute =>{
-    clone.removeAttribute(attribute);
-  });
-
-  const reOpeningTag = /^[\s\S]*?>/;
-  const match = clone.outerHTML.match(reOpeningTag);
+  try {
+    if (ShadowRoot.prototype.isPrototypeOf(element) && element.host && element.localName !== 'a') {
+      element = element.host;
+    }
 
-  return (match && match[0]) || '';
+    const clone = element.cloneNode();
+    ignoreAttrs.forEach(attribute =>{
+      clone.removeAttribute(attribute);
+    });
+    const reOpeningTag = /^[\s\S]*?>/;
+    const match = clone.outerHTML.match(reOpeningTag);
+    return (match && match[0]) || '';
+  } catch (_) {
+    return `<${element.localName}>`;
```

Comment:
can this entire range fail? if it's just the first couple statements, could you reduce the try/catch to just that part?

a comment in the catch block would be useful.